src/map/battle.cpp

```diff
@@ -6810,17 +6810,30 @@ void battle_vanish_damage(struct map_session_data *sd, struct block_list *target
 	nullpo_retv(target);
 
 	// bHPVanishRate
-	int16 vanish_rate_hp = cap_value(sd->bonus.hp_vanish_rate, 0, INT16_MAX);
-	int8 vanish_hp = cap_value(sd->bonus.hp_vanish_per, INT8_MIN, INT8_MAX);
-	// bSPVanishRate
-	int16 vanish_rate_sp = cap_value(sd->bonus.sp_vanish_rate, 0, INT16_MAX);
-	int8 vanish_sp = cap_value(sd->bonus.sp_vanish_per, INT8_MIN, INT8_MAX);
-
-	if (vanish_hp && !(vanish_rate_hp && sd->bonus.hp_vanish_flag & flag && (vanish_rate_hp >= 1000 || rnd() % 1000 < vanish_rate_hp)))
-		vanish_hp = 0;
+	int16 vanish_hp = 0;
+	if (sd->hp_vanish.size()) {
```

Comment:
I would adjust this to:
```suggestion
	if (!sd->hp_vanish.empty()) {
```